atomicapp/nulecule/main.py

```diff
@@ -183,6 +184,11 @@ def run(self, cli_provider, answers_output, ask,
         # it does exist it will be just be loaded and returned
         self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
 
+        # If no provider is specified within answers.conf or --provider=
+        # we should try and get a default one from the artifacts folder
+        if self.answers[GLOBAL_CONF][PROVIDER_KEY] is None and cli_provider is None:
+            self.answers[GLOBAL_CONF][PROVIDER_KEY] = Utils.getDefaultArtifact(self.app_path)
+
```

Comment:
I believe this would be better served by simply leaving the default as it was set in the DEFAULT_ANSWERS and calling out to a function that returns the list of supported providers. If that list has a length of 1 then just set that provider:

```
# comment about single provider means we should default to it
supportedproviders = Utils.getAppSupportedProviders(self.app_path)
if cli_provider is None and len(supportedproviders) ==1:
    self.answers[GLOBAL_CONF][PROVIDER_KEY] = supportedproviders[0]
```